staging/src/k8s.io/apiserver/pkg/server/healthz.go

```diff
@@ -33,10 +33,8 @@ func (s *GenericAPIServer) AddHealthzChecks(checks ...healthz.HealthzChecker) er
 }
 
 // AddReadyzChecks allows you to add a HealthzCheck to readyz.
+// premise: readyzLock has been obtained
```

Comment:
I don't think this is a valid premise. This is a public method which means this can be called arbitrarily during apiserver boot-sequencing. While it is only currently invoked only by `AddDelayedHealthzChecks`, this isn't necessarily true in the future. It is quite likely there will be additional readiness checks which will be added post hoc, and it seems bad to make everyone who decides to use that API to explicitly lock first.